generator/restaurants_info/restaurants_info.cpp

```diff
@@ -0,0 +1,180 @@
+#include "generator/feature_builder.hpp"
+#include "generator/osm_source.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/ftypes_matcher.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+#include "geometry/mercator.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/fstream.hpp"
+#include "std/sstream.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/vector.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+DEFINE_string(osm, "", "Input .o5m file");
+DEFINE_string(out, "", "Output file path");
+
+namespace
+{
+class IsFoodChecker : public ftypes::BaseChecker
+{
+public:
+  static IsFoodChecker const & Instance()
+  {
+    static const IsFoodChecker instance;
+    return instance;
+  }
+
+private:
+  IsFoodChecker()
+  {
+    Classificator const & c = classif();
+    char const * const paths[][2] = {
+      {"amenity", "cafe"},
+      {"amenity", "bar"},
+      {"amenity", "pub"},
+      {"amenity", "fast_food"},
+      {"amenity", "restaurant"}
+    };
+    for (auto const & path : paths)
+      m_types.push_back(c.GetTypeByPath({path[0], path[1]}));
+  }
+};
+
+class Emitter : public EmitterBase
+{
+public:
+  Emitter(vector<FeatureBuilder1> & features)
+    : m_features(features)
+  {
+    LOG_SHORT(LINFO, ("OSM data:", FLAGS_osm));
+  }
+
+  void operator()(FeatureBuilder1 & fb) override
+  {
+    if (!IsFoodChecker::Instance()(fb.GetParams().m_Types) || fb.GetParams().name.IsEmpty())
+    {
+      ++m_stats.m_unexpectedFeatures;
+      return;
+    }
+
+    switch (fb.GetGeomType())
+    {
+    case feature::GEOM_POINT: ++m_stats.m_restaurantsPoi; break;
+    case feature::GEOM_AREA: ++m_stats.m_restaurantsBuilding; break;
+    default: ++m_stats.m_unexpectedFeatures;
+    }
+    m_features.emplace_back(fb);
+  }
+
+  void GetNames(vector<string> & names) const override
+  {
+    names.clear();
+  }
+
+  bool Finish() override
+  {
+    LOG_SHORT(LINFO, ("Number of restaurans (poi):", m_stats.m_restaurantsPoi,
+                      "Number of restaurans (building):", m_stats.m_restaurantsBuilding,
+                      "Number of restaurants (total):", m_features.size(),
+                      "Number of features that have nothing to do with food or that are invalid:",
+                      m_stats.m_unexpectedFeatures));
+    return true;
+  }
+
+private:
+  vector<FeatureBuilder1> & m_features;
+
+  struct Stats
```

Comment:
Why do you need this struct at all? Is it possible to get rid of struct and just create 3 fields in the Emitter class?